android_splash_screen/android/app/src/main/kotlin/com/example/splash_screen_sample/MainActivity.kt

```diff
@@ -0,0 +1,202 @@
+    }
+
+  }
+
+  override fun onFlutterUiDisplayed(){
+    flutterUIReady = true
+
+    if (initialAnimationFinished) {
+      hideSplashScreenAnimation()
+    }
+  }
+
+  override fun onFlutterUiNoLongerDisplayed(){
+    flutterUIReady = false
+  }
+
+  /**
+   * Hides the splash screen only when the entire animation has finished and the Flutter UI is ready to display
+   */
+  private fun hideSplashScreenAnimation(){
+    val splashView = findViewById(R.id.container) as ConstraintLayout
+    splashView
+      .animate()
+      .alpha(0.0f)
+      .setDuration(SPLASHSCREEN_FINAL_ANIMATION_ALPHA_ANIMATION_DURATION)
+      .setListener(
+        object: Animator.AnimatorListener {
+          override fun onAnimationStart(animation: Animator) {}
+          override fun onAnimationEnd(animation: Animator) {}
+          override fun onAnimationCancel(animation: Animator) {}
+          override fun onAnimationRepeat(animation: Animator) {}
+        })
+  }
+
+  /**
+   * Handles the transition from the splash screen to the application
+   */
+  private fun onSplashScreenExit(splashScreenViewProvider: SplashScreenViewProvider) {
+    val accelerateInterpolator = FastOutLinearInInterpolator()
+    val splashScreenView = splashScreenViewProvider.view
+    val iconView = splashScreenViewProvider.iconView
+
+    // We'll change the alpha of the main view
+    val alpha = ValueAnimator.ofInt(255, 0)
+    alpha.duration = SPLASHSCREEN_ALPHA_ANIMATION_DURATION.toLong()
+    alpha.interpolator = accelerateInterpolator
+
+    // And we translate the icon down
+    val translationY = ObjectAnimator.ofFloat(
+      iconView,
+      View.TRANSLATION_Y,
+      iconView.translationY,
+      splashScreenView.height.toFloat()
+    )
+    translationY.duration = SPLASHSCREEN_TY_ANIMATION_DURATION.toLong()
+    translationY.interpolator = accelerateInterpolator
+
+    // And we play all of the animation together
+    val animatorSet = AnimatorSet()
+    animatorSet.playTogether(alpha)
+
+
+    val root = findViewById<ConstraintLayout>(R.id.container)
+    val set1 = ConstraintSet().apply {
+      clone(this@MainActivity, R.layout.main_activity)
+    }
+    set1.applyTo(root)
+    val set2 = ConstraintSet().apply {
+      clone(this@MainActivity, R.layout.main_activity_2)
+    }
+
+    var transitionStarted = false
+    val autoTransition = AutoTransition().apply {
+      interpolator = AccelerateDecelerateInterpolator()
+    }
+    autoTransition.addListener(object: Transition.TransitionListener {
+      override fun onTransitionEnd(transition: Transition) {
+        initialAnimationFinished = true
+
+        if (flutterUIReady) {
+          hideSplashScreenAnimation()
+        }
+    }
+      override fun onTransitionCancel(transition: Transition){}
+      override fun onTransitionPause(transition: Transition) {}
+      override fun onTransitionResume(transition: Transition) {}
+      override fun onTransitionStart(transition: Transition) {}
+    })
```

Comment:
The spacing and indenting seems off here. Did an auto-formatter do this?

Reply:
Yes, this is the automatic spacing